routes/products.router.js

```diff
@@ -0,0 +1,294 @@
+            }
+        });
+
+// comment
+router.post('/products/:productId/comments', authMiddleware,  async (req, res, next) => {
+    try {
+        const { productId } = req.params;
+        const { content } = req.body
+        const { user } = req;
+
+        if (!content) return res.status(400).json({ message: '댓글을 입력해주세요.' });
+
+        const newComment = await prisma.comment.create({
+            data: {
+                content,
+                productId: parseInt(productId),
+                userId: user.id,
+            },
+        });
+        res.status(201).json(newComment);
+    }   catch (error) {
+        next(error);
+    }
+});
+
+//comment check
+router.get('/products/:productId/comments', async (req, res, next) => {
+    try {
+        const { productId } = req.params;
+        let cursor = req.query.cursor ? parseInt(req.query.cursor): undefined;
+        let limit = parseInt(req.query.limit) || 10;
+
+        const comments = await prisma.comment.findMany({
+            where: { productId: parseInt(productId) },
+            select: { id: true, content: true, createdAt: true, userId: true },
+            orderBy: { createdAt: 'desc' },
+            cursor: cursor ? { id: cursor } : undefined,
+            take: limit,
+            skip: cursor ? 1 : 0,
+        });
+        res.status(200).json(comments);
+    }   catch (error) {
+        next(error);
+    }
+});
+
+// comment modify
+router.patch('/products/comments/:commentId', authMiddleware, async (req, res, next) => {
+    try {
+        const { commentId } = req.params;
+        const { content } = req.body;
+        const { user } = req;
+
+        if (!content) return res.status(400).json({ message: '수정할 내용을 입력해주세요.'});
+
+        const existingComment = await prisma.comment.findUnique({ where: { id: parseInt(commentId) } });
+        if (!existingComment || existingComment.userId !== user.id) {
+            return res.status(403).json({ message: '댓글 수정 권한이 없습니다.' });
+        }
+    
+        const updatedComment = await prisma.comment.update 
+({
+         where: { id: parseInt(commentId) },
+         data: { content },
+     });
+        res.status(200).json(updatedComment);
+    }   catch (error) {
+        next(error);
+    }
+});
+
+// comment delete
+router.delete('/products/comments/:commentId', authMiddleware, async (req, res, next) => {
+    try {
+        const { commentId } = req.params;
+        const { user } = req;
+
+        const existingComment = await prisma.comment.findUnique({ where: { id: parseInt(commentId) } });
+        if (!existingComment || existingComment.userId !== user.id) {
+            return res.status(403).json({ message: '댓글 삭제 권한이 없습니다.' });
+        }
+        
+        await prisma.comment.delete({ where: { id: parseInt(commentId) }});
+        res.status(204).send();
+    }   catch(error) {
+        next(error);
+    }
+});
+
+// 상품 좋아요 API
+router.post('/:productId/like', authMiddleware, async (req, res, next) => {
```

Comment:
`/:articleId/like`와 겹칠 수 있어요. `POST /1234/like`로 요청하는 경우 두 API중 하나만 실행될거에요. 한 메서드에 한 url은 유일해야합니다! `products.router.js` 내 등록된 라우터의 경우 `/products`를 붙여주셔도 좋아요.


```suggestion
router.post('/products/:productId/like', authMiddleware, async (req, res, next) => {
```